README.md

```diff
@@ -79,6 +79,7 @@ The supported options are:
 - `--branch` (**Optional**): The git branch from which to download; the latest commit in the branch will be used. If
   specified, will override `--tag`.
 - `--commit` (**Optional**): The SHA of a git commit to download. If specified, will override `--branch` and `--tag`.
+- `--ref` (**Optional**): The git reference to download. If specified, will override `--commit`, `--branch`, and `--tag`.
```

Comment:
Consider putting this first. And updating most of the examples to use it, as it's simpler, and probably our preferred option.

Reply:
Perfect, sounds good to me.

Reply:
One area worth touching on is `--tag`, should `--ref` be able to do the tag filtering that `--tag` is doing? I'm not quite sure yet how to do that if so, since `--ref` is more generic. Perhaps leave the `--tag` examples alone?

Reply:
IIRC, tags are valid refs—given a ref `xxx`, Git will try it as a tag and a branch or something like that, and you can of course force it to be a tag via `/refs/tags/xxx`—but prob worth testing how Git handles those on the CLI!

Reply:
Agreed. I'm thinking about how it might handle `--ref=>1.0.5`. I haven't looked at that bit of the code too much yet.

Reply:
Oh, I gotcha, the version constraints. Yea, those prob need to be parsed before passing on to Git.

Reply:
I've put this first and also solved the handling of tag constraints.